ui/app/components/clients/sync-bar-chart.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { formatNumbers, formatTooltipNumber } from 'vault/utils/chart-helpers';
+import type { SerializedChartData } from 'vault/client-counts';
+import { tracked } from '@glimmer/tracking';
+import { parseAPITimestamp } from 'core/utils/date-formatters';
+import { format } from 'date-fns';
+
+interface Args {
+  data: SerializedChartData[];
+  dataKey: string;
+  chartTitle: string;
+  chartHeight?: number;
+}
+
+interface ChartData {
+  x: string;
+  y: number | null;
+  tooltip: string;
+  legendX: string;
+  legendY: string;
+}
+
+/**
+ * @module ClientsSyncBarChartComponent
+ * Renders a bar chart of secret syncs over time.
+ *
+ * @example
+ * ```js
+ * <Clients::SyncBarChart
+    @chartTitle="Secret Sync client counts"
+    @data={{this.model}}
+    @dataKey="secret_syncs"
+    @showTable={{true}}
+    @chartHeight={{200}}
+  />
+ * ```
+ */
+export default class ClientsSyncBarChartComponent extends Component<Args> {
+  barWidth = 7;
+
+  @tracked activeDatum: ChartData | null = null;
+
+  get chartHeight() {
+    return this.args.chartHeight || 190;
+  }
+
+  get chartData() {
+    return this.args.data.map((d): ChartData => {
+      const date = parseAPITimestamp(d.timestamp) as Date;
+      const count = (d[this.args.dataKey] as number) ?? null;
+      return {
+        x: format(date, 'M/yy'),
+        y: count,
+        tooltip: count === null ? 'No data' : `${formatTooltipNumber(count)} secret syncs`,
+        legendX: format(date, 'MMM yyyy'),
```

Comment:
To be consistent with the line chart can we update this to be the full month name?
<img width="178" alt="Screenshot 2024-01-17 at 3 20 23 PM" src="https://github.com/hashicorp/vault/assets/68122737/7ff63289-d253-4160-a3db-496ee669a3b6">
```suggestion
        legendX: format(date, 'MMMM yyyy'),